packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -98,29 +101,33 @@ class LandmarkManager {
       return;
     }
 
-    let insertPosition = 0;
-    let comparedPosition = newLandmark.ref.current.compareDocumentPosition(this.landmarks[insertPosition].ref.current as Node);
-      // Compare position of landmark being added with existing landmarks.
-      // Iterate through landmarks (which are sorted in document order),
-      // and insert when a landmark is found that is positioned before the newly added element,
-      // or is contained by the newly added element (for nested landmarks).
-      // https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition
-    while (
-      insertPosition < this.landmarks.length &&
-      ((comparedPosition & Node.DOCUMENT_POSITION_PRECEDING) ||
-        (comparedPosition & Node.DOCUMENT_POSITION_CONTAINS))
-    ) {
-      comparedPosition = newLandmark.ref.current.compareDocumentPosition(this.landmarks[insertPosition].ref.current as Node);
-      insertPosition++;
+
+    // Binary search to insert new landmark based on position in document relative to existing landmarks.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition
+    let start = 0;
+    let end = this.landmarks.length - 1;
+    while (start <= end) {
+      let mid = Math.floor((start + end) / 2);
+      let comparedPosition = newLandmark.ref.current.compareDocumentPosition(this.landmarks[mid].ref.current as Node);
+      let isNewAfterExisting = Boolean((comparedPosition & Node.DOCUMENT_POSITION_PRECEDING) || (comparedPosition & Node.DOCUMENT_POSITION_CONTAINS));
+
+      if (isNewAfterExisting) {
+        start = mid + 1;
+      } else {
+        end = mid - 1;
+      }
     }
-    this.landmarks.splice(insertPosition, 0, newLandmark);
+
+    this.landmarks.splice(start, 0, newLandmark);
   }
 
   public updateLandmark(landmark: Pick<Landmark, 'ref'> & Partial<Landmark>) {
     let index = this.landmarks.findIndex(l => l.ref === landmark.ref);
     if (index >= 0) {
       this.landmarks[index] = {...this.landmarks[index], ...landmark};
-      this.checkLabels(this.landmarks[index].role);
+      if (process.env.NODE_ENV !== 'production') {
```

Comment:
So this actually doesn't survive our build step, we'll need to do some research or ask Devon if he knows how React handles dev mode
![Screen Shot 2022-03-08 at 9 42 43 AM](https://user-images.githubusercontent.com/698229/157294922-d55aefb9-b020-490e-b468-a932764fde1a.png)

Reply:
@ktabors Not really useful in production, so probably better to just skip this check. There are probably other places we could start doing this as well.

Reply:
@devongovett is going to bring this up with the parcel team, so I'll remove this for now.